model/trigger/task.go

```diff
@@ -283,52 +282,67 @@ func (t *taskTriggers) taskFirstFailureInVersionWithName(sub *event.Subscription
 }
 
 func (t *taskTriggers) taskRegression(sub *event.Subscription) (*notification.Notification, error) {
-	if t.data.Status != evergreen.TaskFailed || t.task.Requester != evergreen.RepotrackerVersionRequester {
+	shouldNotify, alert, err := isTaskRegression(t.task)
+	if err != nil {
+		return nil, err
+	}
+	if !shouldNotify {
 		return nil, nil
 	}
-
-	previousTask, err := task.FindOne(task.ByBeforeRevisionWithStatuses(t.task.RevisionOrderNumber,
-		task.CompletedStatuses, t.task.BuildVariant, t.task.DisplayName, t.task.Project))
+	n, err := t.generate(sub)
 	if err != nil {
-		return nil, errors.Wrap(err, "error fetching previous task")
+		return nil, err
+	}
+	return n, errors.Wrap(alert.Insert(), "failed to process regression trigger")
+}
 
+func isTaskRegression(t *task.Task) (bool, *alertrecord.AlertRecord, error) {
+	shouldNotify := false
```

Comment:
Nit, but "isRegression" is a better name for this variable.

Reply:
done